### DISCUSSION TITLE
Is ambiguity for meta for login ui node feature or issue?

### DISCUSSION THREAD

**Discussion by winston0410** [ory/kratos#2005] [Q&A] [answered]
I have a schema like this, which `email` is the identifier for password.

```json
{
  "$id": "https://schemas.ory.sh/presets/kratos/quickstart/email-password/identity.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "type": "object",
  "properties": {
    "traits": {
      "type": "object",
      "properties": {
        "email": {
          "type": "string",
          "format": "email",
          "title": "E-Mail",
          "minLength": 3,
          "ory.sh/kratos": {
            "credentials": {
              "password": {
                "identifier": true
              }
            },
            "verification": {
              "via": "email"
            },
            "recovery": {
              "via": "email"
            }
          }
        }
      },
      "required": ["email"],
      "additionalProperties": false
    }
  }
}
```

When I init and get the login flow, I will have a UI node like this:

```
{
//Omitting part of the code here
attributes: {
  disabled: false
  name: "password_identifier"
  node_type: "input"
  required: true
  type: "text"
  value: ""
}
}
```

But my identifier field is email, so I guess it should return `email` in the `node.attributes.type`? Is the ambiguity by using `text` there deliberate, or the handling of that part need some help?

**Comment by aeneasr** [comment, marked as answer]
Nice find :) This is a limitation of the system right now. The problem is that you could theoretically also have multiple identifiers (username + password) but you would still show it as one form field. So while it might be clear in this case (it's an email) it will be unclear in a different case. That's why we decided, for now, to just have this as a regular text input always. If you want to use an email field, you can add a rule in your code which checks for `name: password_identifier` and then displays an email field instead of a text field!

**Reply by winston0410** [reply]
I see, thank you so much for your detail explanation!:)
